libraries/Microsoft.Bot.Expressions/BuiltInFunctions.cs

```diff
@@ -769,74 +817,10 @@ public static (object value, string error) AccessProperty(object instance, strin
             return (value, error);
         }
 
-        private static void ValidateAccessor(Expression expression)
-        {
-            var children = expression.Children;
-            if (children.Length == 0
-                || !(children[0] is Constant cnst)
-                || cnst.ReturnType != ReturnType.String)
-            {
-                throw new Exception($"{expression} must have a string as first argument.");
-            }
-
-            if (children.Length > 2)
-            {
-                throw new Exception($"{expression} has more than 2 children.");
-            }
-
-            if (children.Length == 2 && children[1].ReturnType != ReturnType.Object)
-            {
-                throw new Exception($"{expression} must have an object as its second argument.");
-            }
-        }
-
-        private static (object value, string error) Accessor(Expression expression, object state)
-        {
-            object value = null;
-            string error = null;
-            object instance;
-            var children = expression.Children;
-            if (children.Length == 2)
-            {
-                (instance, error) = children[1].TryEvaluate(state);
-            }
-            else
-            {
-                instance = state;
-            }
-
-            if (error == null && children[0] is Constant cnst && cnst.ReturnType == ReturnType.String)
-            {
-                (value, error) = AccessProperty(instance, (string)cnst.Value);
-            }
-
-            return (value, error);
-        }
-
-        private static (object value, string error) GetProperty(Expression expression, object state)
-        {
-            object value = null;
-            string error;
-            object instance;
-            object property;
-
-            var children = expression.Children;
-            (instance, error) = children[0].TryEvaluate(state);
-            if (error == null)
-            {
-                (property, error) = children[1].TryEvaluate(state);
-                if (error == null)
-                {
-                    (value, error) = AccessProperty(instance, (string)property);
-                }
-            }
-
-            return (value, error);
-        }
-
-        private static object SetProperty(object instance, string property, object value)
+        public static (object result, string error) SetProperty(object instance, string property, object value)
```

Comment:
I'm confused--I thought IMemory was all about removing this stuff?

Reply:
We keep this to be used in the default adapater (SImpleObjectMemory) when you pass an object in. If you pass an IMemory, this code path won't be hit.